### PR TITLE
refactor(ES6): upgrade APIPlugin to ES6

### PR DESCRIPTION
**Pull request by willmendesneto**
**What kind of change does this PR introduce?**

ES5 => ES6 refactor

**Did you add tests for your changes?**

refactor, all existing tests pass

**If relevant, link to documentation update:**

N/A

**Summary**

Helping in part of the ES6 refactor

**Does this PR introduce a breaking change?**

Nope

### CODE REVIEW

File: lib/APIPlugin.js [hunk 1]
```diff
@@ -19,36 +18,41 @@ var REPLACEMENTS = {
...
+		compiler.plugin("compilation", (compilation, params) => {
+			compilation.dependencyFactories.set(ConstDependency, new NullFactory());
+			compilation.dependencyTemplates.set(ConstDependency, new ConstDependency.Template());
+
+			params.normalModuleFactory.plugin("parser", parser => {
+				Object.keys(REPLACEMENTS).forEach(key => {
+					parser.plugin("expression " + key, expr => {
+						const dep = new ConstDependency(REPLACEMENTS[key], expr.range);
+						dep.loc = expr.loc;
+						this.state.current.addDependency(dep);
```
Comment (sokra) [inline lib/APIPlugin.js]: `this` should be `parser` here

Comment (willmendesneto) [inline lib/APIPlugin.js]: Done! Thanks for the feedback @sokra !

**Comment by willmendesneto** [comment]
@sokra Can you take a look again, please?

**Comment by willmendesneto** [comment]
> Should be squashed IMO otherwise LGTM

Yes, for sure (good catch @chicoxyzzy ). We shouldn't add broken commits inside master branch. I don't know for now if we are not using squash as standard, but if not it's a good idea add it.

**Comment by TheLarkInn** [comment]
I'll take care of sqash.

**Pull request merged**